CRM/Financial/BAO/PaymentProcessor.php

```diff
@@ -184,20 +131,72 @@ public static function del($paymentProcessorID) {
     static::deleteRecord(['id' => $paymentProcessorID]);
   }
 
+  /**
+   * Callback for hook_civicrm_pre().
+   * @param \Civi\Core\Event\PreEvent $event
+   */
+  public static function self_hook_civicrm_pre(\Civi\Core\Event\PreEvent $event) {
+    if ($event->action === 'create') {
+      // If we are creating a new PaymentProcessor and have not specified the payment instrument to use, get the default from the Payment Processor Type.
```

Comment:
This was already the case - but this is actually just another parameter on the next chunk of code

Reply:
This code is not new, it's just moved out of the `create` function

Reply:
Yeah - that was what I meant about 'already the case'